core/include/scipp/core/except.h

```diff
@@ -170,6 +170,17 @@ template <class T> void countsOrCountsDensity(const T &object) {
 
 void SCIPP_CORE_EXPORT validSlice(const Dimensions &dims, const Slice &slice);
 
+template <typename T> void info(T t);
+
+template <typename T>
+void SCIPP_CORE_EXPORT validSlice(const T &coords, Dim key,
+                                  const Slice &slice) {
+  if (!coords.contains(key)) {
```

Comment:
There can be datasets with no coord for a certain dimension. I think this check has to be based on the union of the dimensions of all data items (and maybe also all the dimensions of coords, etc.)?